ScrollableTabBar.js

```diff
@@ -187,6 +187,13 @@ const ScrollableTabBar = React.createClass({
     </View>;
   },
 
+  componentWillReceiveProps(nextProps) {
+    // If the tabs change, force the width of the tabs container to be recalculated
+    if (JSON.stringify(this.props.tabs) != JSON.stringify(nextProps.tabs) && this.state._containerWidth) {
```

Comment:
That's why I used JSON.stringify.  It's a poor man's deep equals.  There are more precise ways to do it but they all take a lot more code and have slower performance.

Regardless, yes, I believe this should be sufficient to ensure that any change will trigger the state to update correctly.